curves/jq255s/src/curves/tests.rs

```diff
@@ -0,0 +1,325 @@
+use crate::Affine;
+use ark_ec::double_odd::Projective;
+use ark_ec::AffineRepr;
+use ark_serialize::CanonicalDeserialize;
+use ark_serialize::CanonicalSerialize;
+use ark_std::UniformRand;
+
+#[test]
+fn decode_encode() {
+    let mut rng = ark_std::test_rng();
+    let a = Affine::rand(&mut rng);
+
+    let mut compressed_bytes = Vec::new();
+    a.serialize_compressed(&mut compressed_bytes).unwrap();
+
+    let mut uncompressed_bytes = Vec::new();
+    a.serialize_uncompressed(&mut uncompressed_bytes).unwrap();
+
+    let a_compressed = Affine::deserialize_compressed(&*compressed_bytes).unwrap();
+    let a_uncompressed = Affine::deserialize_uncompressed(&*uncompressed_bytes).unwrap();
+
+    assert_eq!(a_compressed, a);
+    assert_eq!(a_uncompressed, a);
+}
+
+#[test]
+fn valid_decode() {
```

Comment:
Is there a source for these test vectors? If so could you please add a link here?